icu4j/main/classes/core/src/com/ibm/icu/text/RBBITableBuilder.java

```diff
@@ -1158,18 +1163,18 @@ int  getTableSize()  {
                RBBIStateDescriptor sd = fDStates.get(state);
                int row = state*rowLen;
                if (use8Bits) {
-                   Assert.assrt (-128 < sd.fAccepting && sd.fAccepting <= MAX_STATE_FOR_8BITS_TABLE);
-                   Assert.assrt (-128 < sd.fLookAhead && sd.fLookAhead <= MAX_STATE_FOR_8BITS_TABLE);
+                   Assert.assrt (0 <= sd.fAccepting && sd.fAccepting <= 255);
+                   Assert.assrt (0 <= sd.fLookAhead && sd.fLookAhead <= 255);
                } else {
-                   Assert.assrt (-32768 < sd.fAccepting && sd.fAccepting <= 32767);
-                   Assert.assrt (-32768 < sd.fLookAhead && sd.fLookAhead <= 32767);
+                   Assert.assrt (0 <= sd.fAccepting && sd.fAccepting <= 0xffff);
+                   Assert.assrt (0 <= sd.fLookAhead && sd.fLookAhead <= 0xffff);
                }
                table.fTable[row + RBBIDataWrapper.ACCEPTING] = (char)sd.fAccepting;
                table.fTable[row + RBBIDataWrapper.LOOKAHEAD] = (char)sd.fLookAhead;
-               table.fTable[row + RBBIDataWrapper.TAGIDX]    = (char)sd.fTagsIdx;
+               table.fTable[row + RBBIDataWrapper.TAGSIDX]   = (char)sd.fTagsIdx;
                for (col=0; col<numCharCategories; col++) {
                    if (use8Bits) {
-                       Assert.assrt (sd.fDtran[col] <= MAX_STATE_FOR_8BITS_TABLE);
+                       Assert.assrt (0 <= sd.fDtran[col] && sd.fDtran[col] <= MAX_STATE_FOR_8BITS_TABLE);
```

Comment:
I think it should just be
Assert.assrt (0 <= sd.fDtran[col])
Assert.assrt ( sd.fDtran[col] <= MAX_STATE_FOR_8BITS_TABLE);

Reply:
I'm not quite sure what you are intending.
But, I think all the use8Bits asserts could be moved out of this function, since the RBBIDataWrapper form of the table now has 16 bit data always. But the asserts for 0 <= val && val <= 0xffff should remain, since we're moving an int value from the table builder into a char.

RBBIStateTable.put() could then have the 8 bit range asserts, since that's where the 16 bit data gets sliced down to 8.

Reply:
works for me